### PR TITLE
No hooks

### PR DESCRIPTION
**Pull request by Kevinfante**
Implemented the framework for the reviews and ratings sections
-probably should have been done in smaller pull request so sorry about that 
-implemented basic reviews and ratings with some functionality 
- need to go back to refactor the stars component so that it is more accessible to the rest of the project
- need to implement css after this aswell

### CODE REVIEW

**Review by jdougross — APPROVED** [review]

File: client/src/components/Ratings_Reviews/Ratings_Reviews.jsx [hunk 1]
```diff
@@ -1,16 +1,62 @@
...
+    this.getReviews();
+  }
+
+  getMeta() {
+    axios.get(`api/reviews/meta/${this.state.productID}`)
+      .then((res) => {
+        this.setState({ meta: res.data });
+      })
+      .catch((err) => {
+        console.log('err');
```
Comment (jdougross) [inline client/src/components/Ratings_Reviews/Ratings_Reviews.jsx]: could you comment out console logs?  In these cases, could you do res.send(err) instead?

File: client/src/components/Ratings_Reviews/Ratings_Reviews.jsx [hunk 2]
```diff
@@ -1,16 +1,62 @@
...
+      });
+  }
+
+  getReviews() {
+    axios.get(`api/all_reviews/${this.state.sortBy}/${this.state.productID}`)
+      .then((res) => {
+        this.setState({ reviews: res.data });
+      })
+      .catch((err) => {
+        console.log('err');
```
Comment (jdougross) [inline client/src/components/Ratings_Reviews/Ratings_Reviews.jsx]: could you comment out console logs?  In these cases, could you do res.send(err) instead?

File: client/src/components/Ratings_Reviews/ReviewsList/ReviewList.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import ReviewListItem from './ReviewListItem';
+import styles from '../Ratings.module.css';
+
+function ReviewList({ data }) {
+  let datas = [];
+  if (data) {
+    datas = data;
+    console.log('datas: ', data);
```
Comment (jdougross) [inline client/src/components/Ratings_Reviews/ReviewsList/ReviewList.jsx]: comment out console logs?

File: client/src/components/Ratings_Reviews/ReviewsList/photo.jsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+
+function Photos( {pic} ) {
+  console.log('pic: ', pic.url);
```
Comment (jdougross) [inline client/src/components/Ratings_Reviews/ReviewsList/photo.jsx]: comment out console logs?

File: client/src/components/Ratings_Reviews/StatsTab/Characteristics.jsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      </div>
+    )
+  }
+  return returnJSX;
+}
+
+function Characteristics({metaData}) {
+  let barList = [];
+  if (metaData.characteristics) {
+    console.log('characteristics: ', metaData.characteristics);
```
Comment (jdougross) [inline client/src/components/Ratings_Reviews/StatsTab/Characteristics.jsx]: comment out console logs?

File: client/src/components/Ratings_Reviews/StatsTab/RatingBars.jsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    // in return will loop throught and post every bar
+  }
+  return returnJSX;
+};
+
+
+function RatingBars({ metaData }) {
+  let barList = [];
+  if (metaData.ratings) {
+    console.log('ratings: ', metaData.ratings);
```
Comment (jdougross) [inline client/src/components/Ratings_Reviews/StatsTab/RatingBars.jsx]: comment out console logs?

File: client/src/components/Ratings_Reviews/StatsTab/recommended.jsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+const recPer = (obj) => {
+  console.log(obj.true);
```
Comment (jdougross) [inline client/src/components/Ratings_Reviews/StatsTab/recommended.jsx]: comment out console logs?

File: client/src/components/Shared/Stars.jsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    totalSum += ((Object.keys(obj)[i]) * Object.values(obj)[i]);
+  }
+  if ((totalSum / totalCount).isNaN) {
+    return 0;
+  }
+  return totalSum / totalCount;
+};
+
+function Stars({ metaData }) {
+  console.log('metaData: ', metaData);
```
Comment (jdougross) [inline client/src/components/Shared/Stars.jsx]: comment out console logs?

File: client/src/components/Shared/StarsRedo.jsx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+  if ((totalSum / totalCount).isNaN) {
+    return 0;
+  }
+  return totalSum / totalCount;
+};
+
+function Stars({ metaData }) {
+  const[rating, setRating] = useState(4);
+  const[hover, setHover] = useState(null);
+  console.log('metaData: ', metaData);
```
Comment (jdougross) [inline client/src/components/Shared/StarsRedo.jsx]: comment out console logs?

File: client/src/components/Shared/StarsRedo.jsx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+        const ratingValue = i + 1;
+        return (
+        <label>
+          <input
+            type='radio'
+            name='rating'
+            className={styles.starButton}
+            value = {ratingValue}
+            onClick={() => {
+              console.log('clicked')
```
Comment (jdougross) [inline client/src/components/Shared/StarsRedo.jsx]: comment out console logs?

File: server/index.js [hunk 11]
```diff
@@ -40,16 +40,20 @@ app.get('/api/related/', (req, res) => {
...
-      product_id: req.query.product_id,
-    },
+    url: `${baseUrl}/reviews/meta/?product_id=${req.params.productID}`,
+    headers,
+  }).then((axiosResponse) => res.status(200).send(axiosResponse.data))
+    .catch((err) => res.status(400).send());
+});
+
+app.get('/api/all_reviews/:sort/:productID', (req, res) => {
+  console.log('here')
```
Comment (jdougross) [inline server/index.js]: comment out console logs?

**Review by jdougross — CHANGES_REQUESTED** [review]
Hey, nice work.  I think you should comment out the console logs though - when the rest of us pull this component into our main branches and start debugging, console logs we didn't write risk becoming confusing.  ...unsure if I've done this in my requests, but if so, I'll fix it!

**Comment by Kevinfante** [comment]
That's a good point, I've gone ahead and removed all of the console.logs on the file now and made sure to push that commit. Please let me know if it's good to merge now. Thank you.

**Review by jdougross — APPROVED** [review]
Nice work!

**Pull request merged**